File: 8_kyu/Python/descending_order.py
def descending_order(num):
	if num > 0:
		# convert num (integer) into a string so it can be sorted
		toString = str(num)
		# print(toString)
		# print(type(toString))
		# sort the string in reverse (descending) order
		sortStr = sorted(toString, reverse = True)
		# print(type(sortStr))
		# the sorted method converted the string to a list
		# convert the list back to a string
		backToStr = ""
		for digit in sortStr:
			backToStr = backToStr + digit
		# finally return the string converted back into an integer
		return int(backToStr)
	else:
		return 0



print(descending_order(51))
print(descending_order(123456789))

File: 8_kyu/JavaScript/descending_order.js
function descOrder(num) {
	//return int("".join(sorted(str(num), reversed = true)))
	let string = num.toString();
	//return typeof(string);
	//return string;
	let split = string.split("");
	let sortStr = split.sort(function(a,b) {
		return b - a;
	});
	let rejoin = sortStr.join("");
	let toInt = parseInt(rejoin)
	return toInt;
}

console.log(descOrder(42145));


// simplified
function descendingOrder(num) {
	return parseInt(num
		.toString()
		.split("")
		.sort((a,b) => b - a)
		.join(""));
	//return parseInt(result);
}

console.log(descendingOrder(42145));